src/transformState.js

```diff
@@ -5,7 +5,32 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (let i = 0; i < actions.length; i++) {
+    const { type } = actions[i];
+
+    switch (type) {
+      case 'addProperties':
+        for (const [key, value] of Object.entries(actions[i].extraData)) {
+          state[key] = value;
+        }
+        break;
+
+      case 'removeProperties':
+        for (const key of actions[i].keysToRemove) {
+          if (state[key]) {
+            delete state[key];
+          }
+        }
+        break;
+
+      default:
+        if (state) {
+          for (const key of Object.keys(state)) {
+            delete state[key];
+          }
+        }
```

Comment:
It's not a default case, it has its own condition.
Default case can return error message about broken inputs.